pkg/controller/controller.go

```diff
@@ -182,8 +186,57 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 		if utilfeature.DefaultFeatureGate.Enabled(scfeatures.AsyncBindingOperations) {
 			createWorker(c.bindingPollingQueue, "BindingPoller", maxRetries, false, c.requeueServiceBindingForPoll, stopCh, &waitGroup)
 		}
+
+		//createWorker(c.ClusteIDConfigMapQueue, "ClusteIDConfigMap", maxRetries, true, c.reconcileClusteIDConfigMapKey, stopCh, &waitGroup)
 	}
 
+	func() {
+		waitGroup.Add(1)
+		// Cannot wait for the informer to push something into a queue.
+		// What we're waiting on may never exist without us configuring
+		// it, so we have to poll/ ask for it the first time to get it set.
+
+		// Can we ask 'through' an informer? Is it a writeback cache? I
+		// only ever want to monitor and be notified about one configmap
+		// in a hardcoded place.
+		go func() {
+			wait.Until(func() {
+				glog.V(9).Info("cluster ID monitor loop enter")
+				cm, err := c.kubeClient.CoreV1().ConfigMaps("default").Get("cluster-info", metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					m := make(map[string]string)
+					m["id"] = string(uuid.NewUUID())
+					cm := &corev1.ConfigMap{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "cluster-info",
+						},
+						Data: m,
+					}
+					c.kubeClient.CoreV1().ConfigMaps("default").Create(cm)
+					// if we fail to set the id,
+					// it could be due to permissions
+					// or due to being already set while we were trying
+					glog.Warning("could not set clusterid configmap", cm)
+				} else {
+					// cluster id exists and is set
+					// get id out of cm
+					if id, ok := cm.Data["id"]; ok {
+						// set it if it has not been set
+						if "" == c.clusterID {
+							c.clusterID = id
+						} else if id != c.clusterID {
+							glog.Warningf("got a different cluster id than what was stored. controller had %q, got %q.", c.clusterID, id)
```

Comment:
Until we have more discussions I would prefer it we changed c.clusterID rather than ignore the user's request. Or if,after we chat more, we decide to not allow the user to change it once its set then we should reset what's in the CM to match the static c.clusterID.

Reply:
I wish we could do some kind of spec and status thing here, but it's a configmap. it is what it is. since we have a configmap. that's configmap semantics. so I agree that a diff should be adjusted as if it was the value of the configmap.